api/v1_users_authorized_apps.go

```diff
@@ -0,0 +1,77 @@
+package api
+
+import (
+	"time"
+
+	"bridgerton.audius.co/api/dbv1"
+	"bridgerton.audius.co/trashid"
+	"github.com/gofiber/fiber/v2"
+)
+
+type GetUsersAuthorizedAppsQueryParams struct {
+	Limit  int `query:"limit" default:"50" validate:"min=1,max=100"`
+	Offset int `query:"offset" default:"0" validate:"min=0"`
+}
+
+type UserAuthorizedApp struct {
+	Address        string     `json:"address"`
+	Name           string     `json:"name"`
+	Description    *string    `json:"description"`
+	ImageUrl       *string    `json:"image_url"`
+	GrantorUserID  string     `json:"grantor_user_id"`
+	GrantCreatedAt *time.Time `json:"grant_created_at"`
+	GrantUpdatedAt *time.Time `json:"grant_updated_at"`
+}
+
+func (app *ApiServer) v1UsersAuthorizedApps(c *fiber.Ctx) error {
+	queryParams := GetUsersAuthorizedAppsQueryParams{}
+	if err := app.ParseAndValidateQueryParams(c, &queryParams); err != nil {
+		return err
+	}
+
+	// Get authorized apps for the user
+	rows, err := app.queries.GetDeveloperAppsWithGrants(c.Context(), dbv1.GetDeveloperAppsWithGrantsParams{
+		UserID: trashid.HashId(app.getUserId(c)),
+	})
+	if err != nil {
+		return err
+	}
+
+	// Convert to response format and apply pagination
+	var authorizedApps []UserAuthorizedApp
+	start := queryParams.Offset
+	end := queryParams.Offset + queryParams.Limit
+
+	for i, row := range rows {
```

Comment:
Wait so we fetch everything and then throw some away?
Just curious why we're doing processing here instead of using SQL for it?

Reply:
none are thrown away, this is just mapping to the output struct! :)

Reply:
updated the for loop to be friendlier

Reply:
oh the comment was confusing because it says it's doing the pagination